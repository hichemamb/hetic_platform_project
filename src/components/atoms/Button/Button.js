import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const UIColors = {
  primary: {
    color: "transparent",
  },
  secondary: {
    color: "#F44949",
  },
  tertiary: {
    color: "green",
  },
};
const UISizes = {
  small: {
    width: "133px",
  },
  medium: {
    width: "160px",
  },
  large: {
    width: "200px",
  },
};

const Button = ({ size, children, type, onClick, testid }) => {
  return (
    <StyledButton
      size={size}
      type={type}
      onClick={onClick}
      data-testid={testid}
    >
      {" "}
      {children}
    </StyledButton>
  );
};

const StyledButton = styled.button`
  font-size: 16px;
  width: ${(props) => UISizes[props.size].width};
  outline: none;
  transition: all ease 0.4s;
  border: 1px solid #ffffff;
  padding-top: 13px;
  padding-bottom: 13px;
  border-radius: 2px;
  border-color: none;
  color: #ffffff;
  background-color: ${(props) => UIColors[props.type].color};
  cursor: pointer;
  text-transform: uppercase;
  :hover {
    background-color: white;
    color: #2f2f2f;
  }
`;

Button.displayName = "Button";
Button.defaultProps = {
  size: "small",
  type: "primary",
};

Button.propTypes = {
  size: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  testid: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
