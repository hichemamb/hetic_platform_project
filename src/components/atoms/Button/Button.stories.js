import React from 'react';
import Button from './Button';

export const Small = () => (
  <Button size="small" type="primary" onClick={ onClickButton } testid="primary-button">Medium</Button> 
);
export const Medium = () => (
  <Button size="medium"  type="primary" onClick={ onClickButton} >Medium</Button> 
);
export const Large = () => (
  <Button size="large" type="primary" onClick={ onClickButton } >Medium</Button> 
);

const onClickButton = (e) => {
  e.preventDefault();
  alert('Hello button')
}
