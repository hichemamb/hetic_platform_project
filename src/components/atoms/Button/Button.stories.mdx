import { Meta, Story, Preview, Code, Props } from '@storybook/addon-docs/blocks';
import { Small, Medium, Large } from './Button.stories';
import Button from './Button'


<Meta title="Atoms | Button" component={Small, Medium, Large} />

# Button

This docs talk about all variant of `Button` component.  You can define color type with: `Primary`, `Secondary` and `Outlined` & define size type with `Small`, `Medium` and `Large`.
Check the code in clicked to Show code for more details. 

<Preview>
  <Story 
    name="Small default"
    >
    <div style={{ display: 'flex', justifyContent: 'flex-start' }}>
      <Small 
        size="small"
        type="primary"
        onClick={()=>alert("hello small button")}/>
    </div>
  </Story>
</Preview>

<Preview>
  <Story 
    name="Medium"
    >
    <div style={{ display: 'flex', justifyContent: 'flex-start' }}>
      <Medium
        size="medium"
        type="primary"
        onClick={()=>alert("hello medium button")}/>
    </div>
  </Story>
</Preview>


<Preview>
  <Story 
    name="Large"
    >
    <div style={{ display: 'flex', justifyContent: 'flex-start' }}>
      <Large 
        size="large"
        type="primary"
        onClick={()=>alert("hello large button")}/>
    </div>
  </Story>
</Preview>

Copied this code directly !

  ```javascript
  <Button 
    size="small"
    type="primary"
    onClick={()=>alert("hello small button")} />
  ```

# Props

<Props of={Button} />