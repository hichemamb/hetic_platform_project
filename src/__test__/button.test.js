import React from 'react';
import { render } from '@testing-library/react';
import { Small, Medium, Large } from '../components/atoms/Button/Button.stories';
import Button from '../components/atoms/Button/Button.js';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

// Test of Button stories
describe('Renders Button component', () => {
    test('with a default behavior', () => {
    const { container } = render(<Small />);
    expect(container).toBeTruthy();
    });
    test('with a medium size', () => {
    const { container } = render(<Medium />);
    expect(container).toBeTruthy();
    });
    test('with a large size', () => {
    const { container } = render(<Large />);
    expect(container).toBeTruthy();
    });

    // I verify if that he rendering a button style defined with styled components.
    // Each time there is an error this code will print the error 
    // in a snapshots file as well as the sudden style change 
    // of the processed component.
    test('it works', () => {
        const button = renderer.create(<Button />).toJSON()
        expect(button).toMatchSnapshot()
    })
});

