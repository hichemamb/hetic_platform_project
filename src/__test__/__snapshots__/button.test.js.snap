// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button component it works 1`] = `
.c0 {
  font-size: 16px;
  width: 133px;
  outline: none;
  -webkit-transition: all ease .4s;
  transition: all ease .4s;
  border: 1px solid #FFFFFF;
  padding-top: 13px;
  padding-bottom: 13px;
  border-radius: 2px;
  border-color: none;
  color: #FFFFFF;
  background-color: transparent;
  cursor: pointer;
  text-transform: uppercase;
}

.c0:hover {
  background-color: white;
  color: #2F2F2F;
}

<button
  className="c0"
  size="small"
  type="primary"
>
   
   
</button>
`;
